src/NATS.Client.Core/INatsConnection.cs

```diff
@@ -1,7 +1,24 @@
 namespace NATS.Client.Core;
 
-public interface INatsConnection
+public interface INatsConnection : IAsyncDisposable
 {
+    event EventHandler<string>? ConnectionDisconnected;
+
+    event EventHandler<string>? ConnectionOpened;
+
+    event EventHandler<string>? ReconnectFailed;
+
+    event EventHandler<INatsError>? OnError;
+
+    INatsServerInfo? ServerInfo { get; }
+
+    NatsOpts Opts { get; }
+
+    NatsConnectionState ConnectionState { get; }
+
+    NatsHeaderParser HeaderParser { get; }
```

Comment:
This should've been internal perhaps.

Reply:
Should we just make it internal now? It's a breaking change but hopefully there aren't many people relying on it (yet) and there are more breaking changes in this PR anyways.

Reply:
`HeaderParser` should be internal